### PR TITLE
[Notifier] Add Documentation for Events

### PR DESCRIPTION
**Pull request by ismail1432**
I would like to add documentation for the events available in the Notifier, see https://github.com/symfony/symfony-docs/issues/15589

I don't know if I have the good approach (create a specific page like for [console events](https://symfony.com/doc/current/components/console/events.html))

### CODE REVIEW

File: notifier/events.rst [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+The Transport class of the Notifier component allows you to optionally hook
+into the lifecycle via events. Instead of reinventing
+the wheel, it uses the Symfony EventDispatcher component to do the work::
+
+
+The ``MessageEvent::class`` Event
+--------------------------
+
+**Typical Purposes**: Doing something before the message is send (like logging
+which message is going to be send), or displaying something about the event
```
Comment (ismail1432) [inline notifier/events.rst]: remove parenthesis

File: notifier/events.rst [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+    single: Notifier; Events
+
+Using Events
+============
+
+The Transport class of the Notifier component allows you to optionally hook
+into the lifecycle via events. Instead of reinventing
+the wheel, it uses the Symfony EventDispatcher component to do the work::
+
+
```
Comment (ismail1432) [inline notifier/events.rst:9]: check 2 why BL

**Comment by 94noni** [comment]
I would rathee document it like workflow component events (sub part of doc and not new page)
Nice addition :)

**Comment by welcoMattic** [comment]
I guess we need to link your new page here https://symfony.com/doc/current/notifier.html#learn-more

**Comment by ismail1432** [comment]
>I guess we need to link your new page here https://symfony.com/doc/current/notifier.html#learn-more

Thanks @welcoMattic, from what I understand the instruction `notifier/*` [in file](https://github.com/symfony/symfony-docs/blob/5.4/notifier.rst#learn-more) means all pages in the notifier folder will be automatically added so nothing to do specifically.

```
Learn more
----------

.. toctree::
    :maxdepth: 1
    :glob:

    notifier/*
```

**Comment by welcoMattic** [comment]
@javiereguiluz can you review this one please? I guess it's ok, but I'm not RST master :smile:

File: notifier/events.rst [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    single: Notifier; Events
+
+Using Events
+============
+
+The Transport class of the Notifier component allows you to optionally hook
+into the lifecycle via events.
+
+The ``MessageEvent::class`` Event
+--------------------------
```
Comment (OskarStark) [inline notifier/events.rst]: Underline to short

File: notifier/events.rst [hunk 4]
```diff
@@ -0,0 +1,74 @@
+.. index::
+    single: Notifier; Events
+
+Using Events
+============
+
+The Transport class of the Notifier component allows you to optionally hook
```
Comment (OskarStark) [inline notifier/events.rst]:
```suggestion
The class:`...\\..\\Transport` class of the Notifier component allows you to optionally hook
```

File: notifier/events.rst [hunk 5]
```diff
@@ -0,0 +1,74 @@
...
+:class:`Symfony\\Component\\Notifier\\Event\\SentMessageEvent` event::
+
+    use Symfony\Component\Notifier\Event\SentMessageEvent;
+
+    $dispatcher->addListener(SentMessageEvent::class, function (SentMessageEvent $event) {
+        // gets the message instance
+        $message = $event->getOriginalMessage();
+
+        // log something
+        $this->logger(sprintf("The message has been successfully sent and have id: %s, $message->getMessageId()"));
```
Comment (OskarStark) [inline notifier/events.rst]:
```suggestion
        $this->logger(sprintf('The message has been successfully sent and have id: %s, $message->getMessageId()'));
```

File: notifier/events.rst [hunk 6]
```diff
@@ -0,0 +1,74 @@
...
+
+    $dispatcher->addListener(FailedMessageEvent::class, function (FailedMessageEvent $event) {
+        // gets the message instance
+        $message = $event->getMessage();
+
+        // gets the error instance
+        $error = $event->getError();
+
+        // log something
+        $this->logger(sprintf("The message with subject: %s has not been sent successfully. The error is: %s, $message->getSubject(), $error->getMessage()"));
```
Comment (OskarStark) [inline notifier/events.rst]:
```suggestion
        $this->logger(sprintf('The message with subject: %s has not been sent successfully. The error is: %s, $message->getSubject(), $error->getMessage()'));
```

**Comment by ismail1432** [comment]
Hello @OskarStark

Thanks for the review and sorry for the delay I missed the notifications.

**Comment by ismail1432** [comment]
friendly ping 👋

**Review by OskarStark — APPROVED** [review]

**Comment by OskarStark** [comment]
It took time, but here we go, this is in now. Thank you very much Smaine.

**Comment by OskarStark** [comment]
I added a `versionadded` directive after merging

**Comment by ismail1432** [comment]
No worries @OskarStark and big Thanks  for following 👍

**Pull request merged**